### PR TITLE
Fix bluesky-0MQ-proxy CLI on Windows

### PR DESCRIPTION
**Pull request by jklynch**
This PR addresses bluesky issue #1220 

## Description
In setup.py replace `scripts=...` with `entry_points=...`.

The scripts directory has been removed. The old code has been moved to bluesky.commandline.zmq_proxy.py.
 
I prefer the new module since it is more explicit and mirrors caproto.commandline.

## Motivation and Context
`bluesky-0MQ-proxy` installed with `scripts=` in setup.py is not visible in Windows.

## How Has This Been Tested?
A unit test has been added to tests/test_zmq.py. It runs `bluesky-0MQ-proxy` using `subprocess.run` and checks for `returncode == 0.`

I have not been able to run the tests in Windows yet. I spent some time trying to get an Azure Pipeline to run the unit tests but so far that has been a failure.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -14,12 +14,17 @@
     license="BSD (3-clause)",
     url="https://github.com/NSLS-II/bluesky",
     packages=setuptools.find_packages(),
-    scripts=glob.glob('scripts/*'),
+    ##scripts=glob.glob('scripts/*'),
```
Comment (danielballan) [inline setup.py]: Fine to delete this entirely.

Comment (jklynch) [reply setup.py]: I will delete no line before its time.

**Comment by danielballan** [comment]
Oh, might as well delete the scripts directory.

**Comment by mrakitin** [comment]
Is it coordinated with https://github.com/bluesky/bluesky/pull/1114?

**Comment by jklynch** [comment]
I have been trying to run the bluesky tests in Windows, first with VirtualBox and then with an Azure pipeline. So far neither has worked. TravisCI does not have direct support for python on windows. I suppose miniconda could be installed the usual way.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -24,13 +22,31 @@ jobs:
...
+
+  - script: |
+      conda activate bluesky
+      conda install -c anaconda qt sqlite
+      conda install matplotlib pyqt
+    displayName: Install Anaconda packages
+
+  - script: |
+      python -m pip install --upgrade pip
+      pip install -r requirements.txt
```
Comment (danielballan) [inline azure-pipelines.yml]: This line isn't needed; our `setup.py` parses `requirements.txt` and installs them, so `pip install .` is sufficient.

File: azure-pipelines.yml [hunk 3]
```diff
@@ -24,13 +22,31 @@ jobs:
...
+      pip install -r dev-requirements.txt
```
Comment (danielballan) [inline azure-pipelines.yml]: (But this line _is_ needed.)

**Comment by jklynch** [comment]
that CI failure looks kind of random, I don't think it is real

File: azure-pipelines.yml (hunk 2 above)
Comment (jklynch) [reply azure-pipelines.yml]: removed

File: bluesky/commandline/zmq_proxy.py [hunk 4]
```diff
@@ -1,32 +1,31 @@
...
     if logfile:
         set_handler(file=logfile)
+
     def log_writer(name, doc):
         """logger's wrapper function
-            This function will be use to fit .subscribe() method.
-            It has two arguments as .subscribe expect. Inside, it
-            call logger.* to write doc which is a dict as a str
+            This function will be used to fit .subscribe() method.
+            It has two arguments as .subscribe expectd. Inside, it
```
Comment (mrakitin) [inline bluesky/commandline/zmq_proxy.py]:
```suggestion
            It has two arguments as .subscribe expects. Inside, it
```
... or `expected`?

File: bluesky/commandline/zmq_proxy.py [hunk 5]
```diff
@@ -57,11 +56,11 @@ def log_writer(name, doc):
...
         threading.Thread(target=start_dispatcher,
-                            args=(args.logfile,),
-                            daemon=True).start()  # Set daemon to all ipython exit
-                                                  # kill all threads
+                         args=('localhost', out_port, args.logfile),
+                         daemon=True).start()  # Set daemon to all ipython exit
+                                               # kill all threads
     print("Loading...")
-    from bluesky.callbacks.zmq import Proxy  # this takes a couple seconds
+    ##from bluesky.callbacks.zmq import Proxy  # this takes a couple seconds
```
Comment (mrakitin) [inline bluesky/commandline/zmq_proxy.py]: I think it can be removed then.

**Comment by mrakitin** [comment]
@danielballan, are you happy with the changes?

**Pull request merged**